stdlib/LinearAlgebra/src/blas.jl

```diff
@@ -130,6 +128,24 @@ function set_num_threads(n::Integer)
     return nothing
 end
 
+"""
+    get_num_threads()
+
+Get the number of threads the BLAS library is using.
+"""
+function get_num_threads()
+    blas = LinearAlgebra.BLAS.vendor()
+    if blas === :openblas || blas === :openblas64
```

Comment:
replace 
```julia
if blas === :openblas || blas === :openblas64
        return ccall((@blasfunc(openblas_get_num_threads), libblas), Cint, ())
    elseif blas == :mkl
        return ccall((:mkl_get_max_threads, libblas), Cint, ())
    elseif Sys.isapple()
        return Base.parse(Cint, ENV["VECLIB_MAXIMUM_THREADS"])
    else
        error("Unknown BLAS") # better error? return n
```
with
```julia
if blas === :openblas || blas === :openblas64
        return ccall((@blasfunc(openblas_get_num_threads), libblas), Cint, ())
    elseif blas == :mkl
        return ccall((:mkl_get_max_threads, libblas), Cint, ())
   end 
   @static if Sys.isapple()
        return Base.parse(Cint, ENV["VECLIB_MAXIMUM_THREADS"])
    end

    error("Unknown BLAS") # better error? return n
```

Reply:
I don't know if the error would be needed since we covered all cases in `set_num_threads`

Reply:
Ok, can you explain why?

Reply:
`@static if Sys.isapple()` will evaluate the block at parse time when `Sys.isapple()` is true. The code block won't exist when `Sys.isapple()` is false

Reply:
Yeah, but why is this important that the code block disappears at macro expansion time?

Reply:
Is this a performance optimization?

Reply:
It just so just incase the block call code that's only valid for a given OS.